### PR TITLE
File structure

### PR DESCRIPTION
**Pull request by huxuan**
- Basic file structure mostly for `requirements.txt`, `run.py`, `config.sample.py`, `app.__init__` and `app.models`.
- Add basic code style checker and their configuration: `pep8`, `pyflakes`, `pylint`.
- `Makefile` script (with some redundant command but will be useful in the future).
- Minor change, such as `.gitignore`.

P.S. current codebase satisfy `make lint` well.

### CODE REVIEW

**Comment by huxuan** [comment]
Sorry for coming so late. Just some 拖延症……
Feel free to push next time.

**Comment by JeldorPKU** [comment]
Gotcha. I shall review it tonight.

File: run.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+"""
+File: run.py
+Author: huxuan <i(at)huxuan.org>
+Description: Run script for app.
+"""
+import os.path
+
+if not os.path.isfile('config.py'):
+    raise Exception('Please copy `config.sample.py` to `config.py` with proper'
+                    'configuration to make it work.')
```
Comment (JeldorPKU) [inline run.py]: I think you need to add a space between `proper` and `configuration`. LOL

**Comment by JeldorPKU** [comment]
@huxuan Everything looks good except for the missing space where I have put a line note. I will merge it after you add the space there. Good job!

File: run.py (hunk 1 above)
Comment (huxuan) [inline run.py]: OK, got it.

**Pull request merged**